### PR TITLE
[Windows] Adjust min_dimensions and max_dimensions using AdjustWindowRectEx.

### LINKED ISSUE #196: Win32: with_dimensions excludes decoration, with_max_dimensions includes them
I have a reproduction case [here that works with the latest master.](https://gist.github.com/LPGhatguy/8d12e3a9e0fad838a881f200e3687c6c)

It looks like, on Win32 at least, `with_max_dimensions` and `with_min_dimensions` include the window decorations, but `with_dimensions` does not. Only including the client area seems like the correct approach here, so this seems like a bug.

I'm taking a look at this.

### PR DESCRIPTION
**Pull request by Lymia**
This fixes #196, although it does not update the documentation as suggested in the comments.

### CODE REVIEW

**Comment by francesca64** [comment]
Sorry for the wait; you're at the top of my list now. Could you rebase/merge with the current master?

**Comment by Lymia** [comment]
Rebased onto master.

**Comment by Lymia** [comment]
I don't know what's going on with AppVeyor. That looks like a sporadic failure.

File: src/platform/windows/window.rs [hunk 1]
```diff
@@ -622,6 +622,14 @@ pub struct WindowWrapper(HWND, HDC);
 // https://github.com/retep998/winapi-rs/issues/396
 unsafe impl Send for WindowWrapper {}
 
+pub unsafe fn adjust_size(
+    (x, y): (u32, u32), style: DWORD, ex_style: DWORD,
```
Comment (francesca64) [inline src/platform/windows/window.rs]: Ooh, I never knew you could destructure a tuple in the function parameters. Thanks for teaching me something new today!

**Comment by francesca64** [comment]
Thanks! Everything looks good, and it worked great when I tested it on Windows 10.

(I won't fuss about not making the documentation change, since the docs sorta need an overhaul soon anyway.)

**Comment by francesca64** [comment]
...whoops, the CHANGELOG entry was under the wrong section from before. I'll have to remember to correct that.

**Pull request merged**